core/src/main/scala/com/salesforce/op/stages/impl/classification/BinaryClassificationModelSelector.scala

```diff
@@ -206,8 +206,13 @@ case object BinaryClassificationModelSelector {
   ): ModelSelector[ModelType, EstimatorType] = {
     val modelStrings = modelTypesToUse.map(_.entryName)
     val modelsToUse =
-      if (modelsAndParameters == defaultModelsAndParams || modelTypesToUse != modelNames) modelsAndParameters
+      // if no models are specified use the defaults and filter by the named models to use
+      if (modelsAndParameters.isEmpty) defaultModelsAndParams
         .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
+      // if models to use has been specified and the models have been specified filter the models by the names
+      else if (modelTypesToUse != modelNames) modelsAndParameters
+        .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
+      // else just use the specified models
```

Comment:
The presence of comments kind of demonstrates that the code is too quirky.
See what you can do.
First, don't use `modelStrings`. It's misleading. Instead, there must be a method, in `BinaryClassificationModelsToTry`, or, rather, in its superclass (or, better, in a trait) that checks whether it's compatible with `modelTypes. So you can use that method, and write, instead of `modelStrings.contains(e.getClass.getSimpleName)`, `e.isCompatibleWith(modelTypesToUse)`. 
Second, comparing two sequences is weird. If we are dealing with sets, it's one thing; if we are dealing with lists, it's another. Do you mean here a set or a list? What's the reason of this comparison? If these collections are being passed around, it makes more sense to introduce a wrapper class for `Seq[(EstimatorType, Array[ParamMap])]`, call it `ModelsAndParameters`, and throw in functionality into it.
And have a good test coverage. A sequence of `if`s makes the code hard to perceive, and hard to prove that it is actually correct.